stubs/PyAutoGUI/pyautogui/__init__.pyi

```diff
@@ -1,37 +1,177 @@
 
-# These functions reuse pyscreeze functions directly. See above TODO.
-def locate(*args, **kwargs) -> _Box | None: ...
-def locateAll(*args, **kwargs) -> Generator[_Box, None, None]: ...
-def locateAllOnScreen(*args, **kwargs) -> Generator[_Box, None, None]: ...
-def locateCenterOnScreen(*args, **kwargs) -> Point | None: ...
-def locateOnScreen(*args, **kwargs) -> _Box | None: ...
-def locateOnWindow(*args, **kwargs) -> _Box | None: ...
+# These functions reuse pyscreeze functions directly.
+# TODO: Once pyscreeze is typed, we can alias them to simplify this stub
+
+# _locateAll_opencv
+@overload
+def locate(
+    needleImage: str | _Mat,
+    haystackImage: str | _Mat,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat = ...,
+) -> _Box | None: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locate(
+    needleImage: str | Image.Image,
+    haystackImage: str | Image.Image,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> _Box | None: ...
+
+# _locateAll_opencv
+@overload
+def locateAll(
+    needleImage: str | _Mat,
+    haystackImage: str | _Mat,
+    grayscale: bool | None = ...,
+    limit: int = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat = ...,
+) -> Generator[_Box, None, None]: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateAll(
+    needleImage: str | Image.Image,
+    haystackImage: str | Image.Image,
+    grayscale: bool | None = ...,
+    limit: int | None = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> Generator[_Box, None, None]: ...
+
+# _locateAll_opencv
+@overload
+def locateAllOnScreen(
+    image: str | _Mat,
+    grayscale: bool | None = ...,
+    limit: int = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat = ...,
+) -> Generator[_Box, None, None]: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateAllOnScreen(
+    image: str | Image.Image,
+    grayscale: bool | None = ...,
+    limit: int | None = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> Generator[_Box, None, None]: ...
+
+# _locateAll_opencv
+@overload
+def locateCenterOnScreen(
+    image: str | _Mat,
+    minSearchTime: float,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat = ...,
+) -> Point | None: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locateCenterOnScreen(
+    image: str | Image.Image,
+    minSearchTime: float,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: None = ...,
```

Comment:
? Should this not be `SupportsFloat`, like all the other `confidence` parameters?

Reply:
Not for the `_locateAll_python` implementation. Confidence being anything else than `None` will raise an error https://github.com/asweigart/pyscreeze/blob/b693ca9b2c964988a7e924a52f73e15db38511a8/pyscreeze/__init__.py#L246

Reply:
Got it, thanks!